frontend/src/App.tsx

```diff
@@ -186,9 +195,12 @@ export default class App extends React.Component {
     );
   }
 
-  public componentDidMount() {
+  public async componentDidMount() {
     window.addEventListener('keydown', this.onKeyPress);
     window.addEventListener('hashchange', this.onHashChange);
+
+    await this.updateMetadata();
+    setInterval(async () => await this.updateMetadata(), 10000);
```

Comment:
You should use `setTimeout` instead of `setInterval`. Imagine server takes 10s or more to reply. In that case frontend will be generating growing number of requests to the server that is already unable to handle the load, effectively causing DDoS by users.

What should happen instead is this:
* make request to the server
* wait for response (and handle errors, which is not happening right now and will result in uncaught exception
* only then set another timeout try request update 10 seconds later

Reply:
there won't be an exception because there is try-catch inside `utils.fetchMetadata`, let me update the rest

Reply:
Hm... but `fetchMetadata` returns `0` in case there is an error, which is not an object with `uniqueAddressCount` and `spacePledged` fields...

I would have imagined it to throw an error, so you can catch it and just skip `this.appUpdate` call in that case, but at least avoid an error and still show slightly outdated information in the UI.